### PR TITLE
`o3.experimental.FullTensorProductv2` for `torch.compile(...., fullgraph=True)`

### PR DESCRIPTION
**Pull request by mitkotak**
@mariogeiger @Linux-cpp-lisp Would appreciate a review. Also let me know what other tests I can add to `tests/o3/experimental/test_fulltp.py`

Linear coming in next !

### CODE REVIEW

**Comment by mitkotak** [comment]
Here's what the numbers look like on RTX A5500

timing

File: e3nn/o3/__init__.py [hunk 1]
```diff
@@ -57,7 +57,7 @@
 from ._so3grid import SO3Grid
 from ._linear import Linear
 from ._norm import Norm
-
+from .experimental import *
```
Comment (mariogeiger) [inline e3nn/o3/__init__.py:60]: Please import one by one and add in the all list below

File: e3nn/util/datatypes.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+from typing import NamedTuple, Optional
+
+
+class Path(NamedTuple):
```
Comment (mariogeiger) [inline e3nn/util/datatypes.py:4]: I'd call this a Chunk maybe

File: e3nn/util/datatypes.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    mul: int
+    dim: int
+    slice: Optional[slice] = None
+
+
+class Paths(NamedTuple):
```
Comment (mariogeiger) [inline e3nn/util/datatypes.py:10]: And this would be a Path

**Review by mariogeiger — APPROVED** [review]
Hello Mit, looks good👌

**Comment by mitkotak** [comment]
@mariogeiger Thanks for the review ! If all looks good then can you merge ?

**Pull request merged**